### PR TITLE
Document ViewData-backed properties

### LINKED ISSUE #6123: Document ViewData-backed properties
https://docs.microsoft.com/aspnet/core/mvc/views/overview#passing-data-to-views

> Properties decorated with `[ViewData]` on controllers, page models, and Razor Pages provide a convenient way to add data that can be read by views and pages.

> For example, to specify the title for a page and have it show up in the page layout you can define a property on your page model that is decorated with `[ViewData]`:

```csharp
public class AboutModel : PageModel
{
    [ViewData]
    public string Title { get; } = "About";
}
```

> The title can be accessed from the about page as a model property:

```csharp
@page
@model AboutModel

<h2>@Model.Title</h2>
```

> Then, in the layout, the title can be read from the ViewData dictionary:

```cshtml
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>@ViewData["Title"] - WebApplication2</title>
    ...
```

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #6123 

https://review.docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/index?branch=pr-en-us-6148
https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/overview?branch=pr-en-us-6148

### CODE REVIEW

File: aspnetcore/mvc/views/overview.md [hunk 1]
```diff
@@ -118,7 +118,16 @@ Follow the best practice of organizing the file structure for your views to refl
 
 ## Passing data to views
 
-You can pass data to views using several approaches. The most robust approach is to specify a [model](xref:mvc/models/model-binding) type in the view. This model is commonly referred to as a *viewmodel*. You pass an instance of the viewmodel type to the view from the action.
+Pass data to views using several approaches:
+
+* Strongly-typed data: viewmodel
```
Comment (scottaddie) [inline aspnetcore/mvc/views/overview.md]: The hyphen isn't needed when following an "ly" word. Adjust below too.

File: aspnetcore/mvc/views/overview.md [hunk 2]
```diff
@@ -176,12 +185,11 @@ namespace WebApplication1.ViewModels
...
 > Nothing prevents you from using the same classes for both your viewmodel types and your business model types. However, using separate models allows your views to vary independently from the business logic and data access parts of your app. Separation of models and viewmodels also offers security benefits when models use [model binding](xref:mvc/models/model-binding) and [validation](xref:mvc/models/validation) for data sent to the app by the user.
 
-
 <a name="VD_VB"></a>
 
-### Weakly-typed data (ViewData and ViewBag)
+### Weakly-typed data (ViewData, ViewData attribute, and ViewBag)
 
-Note: `ViewBag` isn't available in the Razor Pages.
+Note: `ViewBag` isn't available in Razor Pages.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/overview.md]: I suggest making this a bit more prominent by making it a NOTE widget.

File: aspnetcore/mvc/views/overview.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/mvc/views/overview.md]: 🤔 It's an compound modifier (compound adjective) here tho. As a noun, I would agree ... but as an adjective ... My rubber :duck: says 'hyphen' on this.

File: aspnetcore/mvc/views/overview.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/mvc/views/overview.md]: Ok, I'll change it, but @Rick-Anderson (IIRC here) said that there is some indication that readers read right past our NOTE blocks. If there's more to that story, I wish we had some data to look at ... Are readers *really* just skipping our NOTE notes but actually reading our plain-text notes? Whatever the case with that, I'll add the NOTE blocks here.

Comment (Rick-Anderson) [reply aspnetcore/mvc/views/overview.md]: I prefer not to have notes. Just give them the info without all that distracting chrome.

Comment (Rick-Anderson) [reply aspnetcore/mvc/views/overview.md]: That's a lot of Note chrome on this page. Why convert all the others?

Comment (Rick-Anderson) [reply aspnetcore/mvc/views/overview.md]: I'm OK either way, but I prefer to minimize Note chrome. The security one for sure could use a note.

**Comment by Rick-Anderson** [comment]
>Note The layout is in t

I'd remove that note. It's normal info that doesn't need note chrome.

**Comment by Rick-Anderson** [comment]
chrome is distracting here.
image

File: aspnetcore/mvc/views/overview.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/mvc/views/overview.md]: > I prefer not to have notes. Just give them the info without all that distracting chrome.

That's what I do when I write new topics. I try to just provide the info in-text wherever possible and avoid them generally.

Let's just do the security one. My problem with notes isn't even so much about readers skipping them (although that's a good reason not to use them). I agree that they are generally distracting and unnecessary. Even using the word "Note: " before the note is kind'a silly ...

Note: You need to know this important tidbit.

or

**Note:** You need to know this important tidbit.

I mean every sentence in the topic could have that! lol Emphasis is probably better like this ...

*You need to know this important tidbit.*

... and move on.

> Why convert all the others?

That's what happens when the contractor gets conflicting direction! :smile: lol

This isn't the first time a review has come back 'make a formal NOTE out of that note.'

**Comment by guardrex** [comment]
Try the latest commit as a test of just emphasizing the line. I'm not keen on the word "Note" on these. It's not necessary imo. Everything is a "note." Emphasis can just be mild (italics) or strong (bold) where needed.

**Comment by guardrex** [comment]
@scottaddie U ok with everything here?

**Pull request merged**